### PR TITLE
Image reconstruction based on Grzegorz Korcyl code

### PR DESCRIPTION
**Pull request by grey277**
Adaptation of Grzegorz Korcyl code as UserTask.
Task is using 6 user options: 
ImageReco_CUT_ON_Z_VALUE_float
ImageReco_CUT_ON_LOR_DISTANCE_FROM_CENTER_float
ImageReco_ANNIHILATION_POINT_Z_float
ImageReco_TOT_MIN_VALUE_float
ImageReco_TOT_MAX_VALUE_float
ImageReco_ANGLE_DELTA_MIN_VALUE_float
all cuts was adapted from Monika's EventCategorizer.
It still need to be tested.
edit: should be accepted after: https://github.com/JPETTomography/j-pet-framework/pull/145

### CODE REVIEW

**Comment by grey277** [comment]
I tested it and it seems to work fine, I added also histogram representing how much hits each cut filtered.

**Comment by grey277** [comment]
New version with fixed time units in reconstruction(ps changed to ns).
In that version 1 bin is 8 1/3 mm it cannot be less, because TBufferFile cannot write more then 1073741822 bytes.

**Comment by wkrzemien** [comment]
I accepted https://github.com/JPETTomography/j-pet-framework/pull/145 
So please redirect j-pet-framework to develop and it should pass, the tests

**Comment by grey277** [comment]
Tests are not passing, because some of the download files were removed from sphinx, was it intentional?

File: LargeBarrelAnalysis/ImageReco.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+//when cint is used instead of compiler, override word is not recognized
+//nevertheless it's needed for checking if the structure of project is correct
+#define override
+#endif
+
+#include <JPetUserTask/JPetUserTask.h>
+#include <memory>
+
+/**
+ * @brief TODO...
```
Comment (wkrzemien) [inline LargeBarrelAnalysis/ImageReco.h]: Please, Add two-line description. 
E.g.
Image reconstruction module which should be used for quick, rough data tests.
It implements a simple image reconstruction algorithm adapted from G. Korcyl's code. It is equivalent to one back-projection without any filtering but including TOF information, with the assumption that TOF error =0.  Before reconstructruction, the data are preselected based on the set of conditions defined in the json file such as:
-(enumerate cuts you  are using).
The point of anihilation is reconstructed for all combination of registered hits in given events.

File: LargeBarrelAnalysis/ImageReco.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+{
+
+public:
+  ImageReco(const char* name);
+  virtual ~ImageReco();
+  virtual bool init() override;
+  virtual bool exec() override;
+  virtual bool terminate() override;
+private:
+  bool calculateReconstructedPosition(const JPetHit& firstHit, const JPetHit& secondHit);
```
Comment (wkrzemien) [inline LargeBarrelAnalysis/ImageReco.h]: Add some comment how do you calculate it
Also, this method should be called
calculateVertexPosition
or
calculateAnihillationPoint

File: LargeBarrelAnalysis/ImageReco.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  bool checkConditions(const JPetHit& first, const JPetHit& second);
+
+  const std::string fCutOnZValueKey = "ImageReco_CUT_ON_Z_VALUE_float";
+  const std::string fCutOnLORDistanceKey = "ImageReco_CUT_ON_LOR_DISTANCE_FROM_CENTER_float";
+  const std::string fCutOnAnnihilationPointZ = "ImageReco_ANNIHILATION_POINT_Z_float";
+  const std::string fCutOnTOTMinValue = "ImageReco_TOT_MIN_VALUE_IN_NS_float";
+  const std::string fCutOnTOTMaxValue = "ImageReco_TOT_MAX_VALUE_IN_NS_float";
+  const std::string fCutOnAngleDeltaMinValue = "ImageReco_ANGLE_DELTA_MIN_VALUE_float";
+
+  const int xRange = 50;
```
Comment (wkrzemien) [inline LargeBarrelAnalysis/ImageReco.h]: pls,
use f for fields
and k for consts.

**Comment by grey277** [comment]
I updated code including requested changes and also added 4 new options, that allows to set size of 3d histogram and size of binning.

**Pull request merged**